### DISCUSSION TITLE
when login to github  in goland  show Failed to initiate the GitHub login process. Please try again

### DISCUSSION THREAD

**Discussion by zhuxiaoker** [community/community#16965] [Copilot Conversations]
As title, how to solve it ?

**Comment by zhuxiaoker** [comment]
fix it by downgrade copilot

**Reply by SalHe** [reply]
same with u. fix by downgrade.

**Reply by thouger** [reply]
reply from the eighth reply

**Reply by Xiao-Chenguang** [reply]
> which version ? thanks.

1.1.20 works

**Reply by thexzan** [reply]
how to downgrade copilot ?

**Reply by june1888** [reply]
> how to downgrade copilot ?

I have the same issue, did you figure out how to downgrade copilot

**Reply by thexzan** [reply]
> > how to downgrade copilot ?
> 
> I have the same issue, did you figure out how to downgrade copilot

yup, please see my answer below

**Reply by DarkRepulser1227** [reply]
> how to downgrade copilot ?

https://plugins.jetbrains.com/plugin/17718-github-copilot/versions/stable
check is

**Reply by Moses-Code-Dev** [reply]
This worked for me
https://github.com/github-community/community/discussions/16965#discussioncomment-2916107

**Comment by badboy-tian** [comment]
+1

**Reply by thouger** [reply]
same problem,and don't know what happen，copilot is my depth product experience

**Comment by SummerSec** [comment]
I think it's this problem
![image](https://user-images.githubusercontent.com/47944478/169989664-686f8cbe-06b3-4dc8-ac95-5771ae15c44f.png)

**Reply by lyjwolf318** [reply]
how to fix it?

**Comment by liutao62** [comment]
same problem happens when i updated the plugin

**Comment by Raxcl** [comment]
+1

**Comment by jhao14** [comment]
same problem happens when i updated the plugin +1

**Comment by thouger** [comment]
the newest and worked version is `github-copilot-intellij-1.1.20.1417.zip`,you can download from [https://plugins.jetbrains.com/plugin/17718-github-copilot/versions/stable](url)

**Comment by smithyj** [comment]
me too same error

**Reply by whenk7** [reply]
https://github.com/orgs/github-community/discussions/16965#discussioncomment-2916107

**Comment by thexzan** [comment]
as **thouger** comment above, 
the last working version is **14.17**

- download version **14.17** from [https://plugins.jetbrains.com/plugin/17718-github-copilot/versions/stable](https://github.com/github/feedback/discussions/url)
- go to your IDE ( Intellij and etc ) and go to preferences and then plugin
- Click on gear icon, then install plugin from disk
- Choose downloaded **14.17** file 
- Restart to finish plugin installation then try copilot again

**Reply by june1888** [reply]
hi, this link shows 404 page not found

**Reply by thexzan** [reply]
> hi, this link shows 404 page not found

maybe try copy and paste the link

**Reply by june1888** [reply]
thank you, I download the 14.17 file, and when I Choose downloaded 14.17 file AND Restart to finish plugin installation, but I did not see  GitHub Copilot option in Tools. so weird

**Comment by vanarok** [comment]
I was able to solve the problem in myself(webstorm). Here's what i did: 1. Update plugin copilot to last version. 2. Clicked "Repair IDE". 3. Deleted the authorization file from settings folder IDE(.../JetBrains/WebstormXXXX.X/options/github-copilot.local.xml. 4. Re-authorized in copilot.

**Reply by rosuH** [reply]
Works for me, thanks~

**Comment by MontaukLaw** [comment]
嗯, 我down grade就可以解决了.  I already fixed it by down grade the copilot, worked on CLion

**Reply by Myfanqie** [reply]
Thank you , I also lowered the version，and copilot is OK use

**Comment by mythuscry** [comment]
please , how do you  down grade copilot , do you need install it from web

**Reply by SalHe** [reply]
Choosing a correct version plugin and download from Jetbrains Marketplace, and then install plugin from disk in IDEA or other IDE based on IDEA.

**Reply by SalHe** [reply]
https://plugins.jetbrains.com/

**Comment by smithyj** [comment]
same error, `1.1.24.1610` version is same error

**Reply by mythuscry** [reply]
> same error, `1.1.24.1610` version is same error

hi ，bro , get into  https://plugins.jetbrains.com/ ,you could find lower verision , from disk to install

**Comment by pakchoily** [comment]
在idea使用GitHub账号、Copilot异常
登录GitHub显示
`Invalid authentication data.Connection refused: connect`

或者Copilot显示
`Failed to initiate the GitHub login process. Please try again.`

网上一般的方法都是推荐使用token登录,或者降级copilot
经过研究，其实就DNS污染
修改hosts（‪C:\Windows\System32\drivers\etc\hosts）,添加以下参数。[https://www.ipaddress.com/](https://www.ipaddress.com/) 可以查询最新地址
**一定要添加api.github.com**
**must add ip of api.github.com**
~~~
# Github
140.82.114.3 github.com
199.232.69.194 github.global.ssl.fastly.net
140.82.114.5 api.github.com
~~~
之后cmd 刷新dns
~~~
ipconfig /flushdns
~~~
当浏览器可以裸连https://github.com ,表示dns刷新成功，这样所有问题都解决。

**Reply by smithyj** [reply]
是的。我抓包发现了

**Reply by HackerOQ** [reply]
> 在idea使用GitHub账号、Copilot异常 登录GitHub显示 `Invalid authentication data.Connection refused: connect`
> 
> 或者Copilot显示 `Failed to initiate the GitHub login process. Please try again.`
> 
> 网上一般的方法都是推荐使用token登录,或者降级copilot 经过研究，其实就DNS污染 修改hosts（‪C:\Windows\System32\drivers\etc\hosts）,添加以下参数。https://www.ipaddress.com/ 可以查询最新地址 **一定要添加api.github.com** **must add ip of api.github.com**
> 
> ```
> # Github
> 140.82.114.3 github.com
> 199.232.69.194 github.global.ssl.fastly.net
> 140.82.114.5 api.github.com
> ```
> 
> 之后cmd 刷新dns
> 
> ```
> ipconfig /flushdns
> ```
> 
> 当浏览器可以裸连https://github.com ,表示dns刷新成功，这样所有问题都解决。

修改hosts 
> # Github
> 140.82.114.3 github.com
> 199.232.69.194 github.global.ssl.fastly.net
> 140.82.114.5 api.github.com

这样不太行，根据https://www.ipaddress.com/ 查询最新地址ip后替换哪个IP

**Reply by pakchoily** [reply]
> > 在idea使用GitHub账号、Copilot异常 登录GitHub显示 `Invalid authentication data.Connection refused: connect`
> > 或者Copilot显示 `Failed to initiate the GitHub login process. Please try again.`
> > 网上一般的方法都是推荐使用token登录,或者降级copilot 经过研究，其实就DNS污染 修改hosts（‪C:\Windows\System32\drivers\etc\hosts）,添加以下参数。https://www.ipaddress.com/ 可以查询最新地址 **一定要添加api.github.com** **must add ip of api.github.com**
> > ```
> > # Github
> > 140.82.114.3 github.com
> > 199.232.69.194 github.global.ssl.fastly.net
> > 140.82.114.5 api.github.com
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 之后cmd 刷新dns
> > ```
> > ipconfig /flushdns
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 当浏览器可以裸连https://github.com ,表示dns刷新成功，这样所有问题都解决。
> 
> 修改hosts
> 
> > # Github
> > 140.82.114.3 github.com
> > 199.232.69.194 github.global.ssl.fastly.net
> > 140.82.114.5 api.github.com
> 
> 这样不太行，根据https://www.ipaddress.com/ 查询最新地址ip后替换哪个IP

三个你都可以根据最新的IP来换，我的意思是api.github.com 可能过段时间就不是140.82.114.5了，根据最新查出来的IP填写就好。

**Reply by zhongjun96** [reply]
Recommended [GitHub520](https://github.com/521xueweihan/GitHub520)

**Reply by 77yang** [reply]
6666 亲测有效

**Reply by li19876** [reply]
我的设置了也不管用，浏览器可以裸连github

**Reply by ooeooo** [reply]
解决了，太感谢了！

**Reply by minlik** [reply]
亲测有效，太牛了

**Reply by tvbic** [reply]
> 我的設置了也不管用，瀏覽器可以裸連github

我試了也是沒用

**Reply by a807875765** [reply]
是需要外网IP的 国内的IP都可以裸连github 没啥用

**Reply by XDo0** [reply]
如果有代理的话，开个全局代理，再执行
```
ipconfig /flushdns
```
就能登录成功了

**Comment by MillionMindFU** [comment]
非常有用，谢谢！
Thank you！

**Reply by HackerOQ** [reply]
> 非常有用，谢谢！ Thank you！

你好，直接修改hosts：
140.82.114.3 github.com
199.232.69.194 github.global.ssl.fastly.net
140.82.114.5 api.github.com
这样可以？

**Comment by 0333814695** [comment]
> As title, how to solve it ?

**Reply by Moses-Code-Dev** [reply]
This worked for me
https://github.com/github-community/community/discussions/16965#discussioncomment-2916107

**Comment by Blackjack200** [comment]
在修改了DNS后，我仍然不可用，直到我把bashrc的HTTP_PROXY HTTPS_PROXY ALL_PROXY注释掉

**Reply by yueqing-wolai** [reply]
仍然不可用

**Reply by Myfanqie** [reply]
现在最新版的可以用了，似乎

**Reply by yueqing-wolai** [reply]
> 现在最新版的可以用了，似乎

1.1.24.1610 会报 Failed to initiate the GitHub login process. Please try again.

**Reply by Moses-Code-Dev** [reply]
https://github.com/github-community/community/discussions/16965#discussioncomment-2916107

**Comment by joseserro** [comment]
To anyone who's on windows 7:
They apparently updated node to a more recent version which requires Win 8.1 or higher. To circumvent this decision you can add NODE_SKIP_PLATFORM_CHECK to your environment variables with the value 1, then install the latest nightly build of copilot, go to %appdata%/JetBrains/YOUR_IDE_OF_CHOICE/plugins/github-copilot-intellij/copilot-agent/bin and replace the binaries with the ones from version 1.1.20.1417.
This allows copilot to keep using the old copilot-agent with the latest version of the plugin, so it stops complaining about needing to update.

**Comment by Moses-Code-Dev** [comment]
**100% Working solution** 

1- Update your github-copilot to the latest version (1.1.27.1708) then close the idea  
                                                                                       
2- Download this version of github-copilot (1.1.20.1417):                              
https://plugins.jetbrains.com/plugin/download?rel=true&updateId=172765 and Extract it  
                                                                                       
4- Navigate to ...\github-copilot-intellij-1.1.20.1417\github-copilot-intellij\lib     
                                                                                       
5- Copy "core-1.1.20" file                                                             
                                                                                       
6- Navigate to                                                                         
                                                                                       
* For IntelliJ IDEA                                                                    
...\AppData\Roaming\JetBrains\IdeaIC2022.1\plugins\github-copilot-intellij\lib         
                                                                                       
* For AndroidStudio                                                                    
...\AppData\Roaming\Google\AndroidStudio2021.2\plugins\github-copilot-intellij\lib     
                                                                                       
7- Replace "core-1.1.24" with "core-1.1.20".                                           
                                                                                       
Enjoy...              

**Update 26/06/2022** :          

Use this updated solution instead  

https://github.com/djoxy-code/Copilot_plugin_patch

**Reply by wave1996** [reply]
牛逼兄弟

**Reply by HackerOQ** [reply]
摧古拉斯！！！

**Reply by yueqing-wolai** [reply]
By following the above steps, another error raised from copilot: JsonRpcErrorException(requestId=3, code=-32603, extraData=null)
	at com.github.copilot.lang.agent.rpc.JsonRPC.parseResponse(JsonRPC.java:52)

**Reply by myname-dyh** [reply]
解决+1

**Reply by Moses-Code-Dev** [reply]
@yueqing-wolai, make sure you followed all steps correctly.
https://github.com/github-community/community/discussions/18132#discussion-4130423

**Reply by CreatorEdition** [reply]
> By following the above steps, another error raised from copilot: JsonRpcErrorException(requestId=3, code=-32603, extraData=null) at com.github.copilot.lang.agent.rpc.JsonRPC.parseResponse(JsonRPC.java:52)

你相当于下载一个本地的插件，然后把之前的版本卸载后，重启，最后通过本地安装插件的形式出现安装进去试试

**Reply by wuyi2004** [reply]
完美,牛逼

**Reply by CreatorEdition** [reply]
> 完美,牛逼

哈哈哈，看来真的是哀嚎一片啊

**Reply by weiyuming** [reply]
感谢大佬，

**Reply by weiyuming** [reply]
我的救星啊，现在没这个  感觉不会写代码了

**Reply by ooeooo** [reply]
thanks very much!

**Reply by yueqing-wolai** [reply]
尝试了换不同的 idea， 使用降级版本 1417, 和以上替换 core 的方法，没有一个 work。。。

**Reply by lakkamaa** [reply]
thanks

**Reply by liangzw4116** [reply]
最新版本1.1.25.1678的这个解决方法可以，其他降级使用1.1.20.1417的还是不行的，登录成功后还是会报433错误。其实解决方法一句话总结就是用1.1.20.1417版本的core-1.1.20.jar包替换掉最新版本的core包

**Reply by liangzw4116** [reply]
最新版本1.1.25.1678的这个解决方法可以，其他降级使用1.1.20.1417的还是不行的，登录成功后还是会报433错误。其实解决方法一句话总结就是用1.1.20.1417版本的core-1.1.20.jar包替换掉最新版本的core包

**Reply by ferlini** [reply]
That's useful!
Step 6- Navigate to MacOS JetBrains:
`~/Library/Application Support/JetBrains/RubyMine2022.1/plugins/github-copilot-intellij/lib`

**Reply by hovinlu** [reply]
cool, it worked. thank you

**Reply by quickshare2002** [reply]
OMG，the best answer！！！！

**Reply by whenk7** [reply]
a Perfect Solution!

**Reply by whenk7** [reply]
> a Perfect Solution!

After the replacement, the login status is displayed directly

**Reply by Akitaqaq** [reply]
牛逼

**Reply by sss768** [reply]
一次成功，比代理好用

**Reply by zjyuw** [reply]
Thank you, this solved my problem

**Reply by yinpanhui** [reply]
解决+1

**Reply by mumuxi176** [reply]
解决了，感谢，thanks

**Reply by atticus-lv** [reply]
AMAZING!!!

**Reply by a807875765** [reply]
idea 更新最新版本 替换1.1.26 core的jar包 仍然出现waiting状态

**Reply by s0ing** [reply]
完美

**Reply by Amazingmonkeys** [reply]
bingo from Chinese mainland

**Reply by zhouwentong1993** [reply]
Amazing

**Reply by gclsoft** [reply]
> By following the above steps, another error raised from copilot: JsonRpcErrorException(requestId=3, code=-32603, extraData=null) at com.github.copilot.lang.agent.rpc.JsonRPC.parseResponse(JsonRPC.java:52)

I'm the same. Have you solved it?

**Reply by jusescn** [reply]
idea ->help -> edit vm properties
add -Dcopilot.agent.disabled=true  
用这种方式也能登陆。

**Comment by Deomaning** [comment]
我 github 可以裸连 ,为什么还是出这个问题

**Reply by Moses-Code-Dev** [reply]
https://github.com/github-community/community/discussions/16965?sort=new#discussioncomment-2916107

**Reply by pakchoily** [reply]
> 我 github 可以裸连 ,为什么还是出这个问题

因为dns无法解析 api.github.com，所以手动在hosts 指定ip。

**Comment by Akalanka1337** [comment]
Hey,
Did you tried to login using token?
https://github.com/settings/tokens

**Reply by yueqing-wolai** [reply]
How to login with token, I can't find the option from copilot.

**Reply by CreatorEdition** [reply]
> 如何使用令牌登录，我在副驾驶中找不到选项。

可以尝试根据上面的方式恢复到旧版本，然后离线安装插件

**Reply by zzulpf** [reply]
> 

老哥，离线安装旧版本后IDE会提示当前版本不可用，需要更新插件。。。。

**Reply by CreatorEdition** [reply]
> > 
> 
> 老哥，离线安装旧版本后IDE会提示当前版本不可用，需要更新插件。。。。

先把版本更新到最新版本，然后用离线插件包的 core-1.1.20 变更文件名称为 core-1.1.25，并覆盖到 
...\AppData\Roaming\JetBrains\XXXXXXXXXXXXXXX\plugins\github-copilot-intellij\lib
重启启用插件

**Comment by SakurajimaMaii** [comment]
+1

**Reply by quickshare2002** [reply]
兄弟 爬楼，这个能解决 "https://github.com/orgs/github-community/discussions/16965#discussioncomment-2916107"

**Comment by smithyj** [comment]
The problem is resolved in latest version `1.1.25.1678`

**Reply by dumengzhao** [reply]
这个版本也没有解决

**Reply by sss768** [reply]
It may have solved your own problems, but it still haven't solved the problem that GFW in Chinese Mainland interferes with github.com access. The main reason is that copilot does not automatically use the system agent,  so our  agent software "clash for Windows" cannot let copilot access github.com even if the global agent.

**Reply by z2879878189z** [reply]
I have solved the problem with the latest version in Chinese Mainland,Same version 1.1.25.1678

**Reply by Moses-Code-Dev** [reply]
You can use the Updated method 

https://github.com/djoxy-code/Copilot_plugin_patch

**Reply by dumengzhao** [reply]
按照这个方法可以 -> https://github.com/djoxy-code/Copilot_plugin_patch

**Reply by wave1996** [reply]
> 您可以使用更新的方法
> 
> https://github.com/djoxy-code/Copilot_plugin_patch

![89822D38AB234E4A2EC1836C3E79B89E](https://user-images.githubusercontent.com/33834035/174204018-94b0fc33-5008-4c46-aab5-00eb53bb2b2c.png)

**Reply by zjyuw** [reply]
This version does not solve this problem

**Comment by Hardwaregore** [comment]
Up grade your VScode

**Comment by a807875765** [comment]
尝试了下 各种方法  目前升级到最新版本后之后 替换core的包 无效， 之后更换了host 使用香港节点 一直waiting  更换了美国西雅图的节点 完成 可能不光和host文件有关 还关联了本地的IP

**Reply by CreatorEdition** [reply]
编辑器有添加代理吗，防火墙有没有另外阻断，通过cmd的 curl指令是否能正常联网，替换过程中是否将插件禁用后替换？

**Reply by a807875765** [reply]
插件之前换了core包，还是没用考虑到后面更新问题还是卸载了后重新下载了，编辑器没有使用代理，全局V2ARY还是开了代理的，不开代理连验证码和验证网站都打不开，防火墙没有操作，curl可以正常联网

**Comment by CreatorEdition** [comment]
不是这个问题的，直接更换core包是可以解决的，不需要更换host的情况下，你把v2的局域网功能打开，然后通过设置代码器的代理到v2上面试试

在 2022-06-21 18:04:00，a807875765 ***@***.***> 写道：


尝试了下 各种方法 目前升级到最新版本后之后 替换core的包 无效， 之后更换了host 使用香港节点 一直waiting 更换了美国西雅图的节点 完成 可能不光和host文件有关 还关联了本地的IP

**Reply by a807875765** [reply]
之前没有打开局域网功能 开的是全局 应该和host差不多 就是一直接收不到响应

**Comment by cc970212cc** [comment]
已经要收费了，别折腾了

**Reply by a807875765** [reply]
是的 10刀一个月好像 可以close了

**Comment by ruanrz** [comment]
要收费了，可以选择试用到2022.08.22（注意多了个获取代码的选项）

**Reply by smithyj** [reply]
是的不能白嫖了，他只会学习你的GitHub代码吧，项目代码应该不会学习吧

**Reply by a807875765** [reply]
就是扫描一些代码仓库的公用代码，一断网就不行了，就是替代你自己百度搜索而已

**Comment by henryJhonOt** [comment]
笑死, 一堆中国人一开始全在飙英语, 直到一位老哥用了中文, 下面全是中文

**Reply by codingJourneyFromUnemployment** [reply]
哈哈哈~~

**Comment by anyone0034** [comment]
1.26按照此方法，失败。

**Comment by smithyj** [comment]
The Answer need to closed!

**Comment by PoplarTang** [comment]
我可以了！同志们！用的是360的”断网急救箱“，”全面诊断“后再点击“强力修复”，把所有网卡相关的配置全初始化了就好了。我之前尝试过各种Proxy、VPN代理（Proxifier、fiddler、clash等等）都不行，就这个网络修复一下可以了！

**Comment by TaylorAndTony** [comment]
2022-08-04 亲测可行，改 host，记得现查这三个网址对应的啥IP

140.82.114.3 github.com
199.232.69.194 github.global.ssl.fastly.net
140.82.114.5 api.github.com

回IDE去点右下角机器人标登录即可，还不行的话，试试先在 IDE 里登录 github 账号，设置→版本控制→Github

**Comment by codingJourneyFromUnemployment** [comment]
2022-08-11 各位老哥,我升级了pycharm最新的版本(2022.2), copilot也升级到了最新的版本(1.1.29.1869), 又出现了"Failed to initiate the GitHub login process. Please try again"问题. 尝试了修改dns, 降版本, 360急救箱, 替换core包, 先在ide里面登陆github账号等一切办法, 均无法修复. 请问这病还能治吗?

**Comment by codingJourneyFromUnemployment** [comment]
2022-08-11 hey guys, I have upgraded the latest version of pycharm (2022.2) and copilot to the latest version (1.1.29.1869), and the problem of "Failed to initiate the GitHub login process. Please try again" appeared again . I've tried all methods such as modifying dns, downgrading the version, 360 first aid kit, replacing the core package, logging in to the github account in the ide, etc., but they cannot be repaired. 
Is there any help in this situation?
